File: frontend/src/types/historicalRetrieval.ts
export interface SimilarHistoricalIncident {
  historical_incident_id: string;
  title: string;
  summary: string;
  symptoms: string;
  root_cause: string | null;
  resolution: string | null;
  severity: string;
  service_id: string | null;
  occurred_at: string;
  similarity_score: number;
}

File: backend/app/schemas/historical_retrieval.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class SimilarHistoricalIncidentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    historical_incident_id: UUID
    title: str
    summary: str
    symptoms: str
    root_cause: str | None
    resolution: str | None
    severity: str
    service_id: UUID | None
    occurred_at: datetime
    similarity_score: float